src/main/java/dima/senko/bookstore/repositiry/SpecificationProvider.java

```diff
@@ -0,0 +1,10 @@
+package dima.senko.bookstore.repositiry;
+
+import org.springframework.data.jpa.domain.Specification;
+
+public interface SpecificationProvider<T> {
+
```

Comment:
```suggestion
```
no empty line after method/class definition is required according to mate code style